.changeset/itchy-candles-type.md

```diff
@@ -0,0 +1,30 @@
+---
+'@backstage/plugin-org': patch
+---
+
+Introduced a new MyGroupsSidebarItem SidebarItem that links to one or more groups based on the logged in user's membership.
+
+To use MyGroupsSidebarItem you'll need to add it to your `Root.tsx` like this:
+
+```diff
+// app/src/components/Root/Root.tsx
++ import { MyGroupsSidebarItem } from '@backstage/plugin-org';
++ import GroupIcon from '@material-ui/icons/People';
+
+<SidebarPage>
+    <Sidebar>
+      //...
+      <SidebarGroup label="Menu" icon={<MenuIcon />}>
+        {/* Global nav, not org-specific */}
+        //...
+        <SidebarItem icon={HomeIcon} to="catalog" text="Home" />
++       <MyGroupsSidebarItem props={{
```

Comment:
What's up with this `props={{` thing? Aren't we supposed to call this as

```jsx
          <MyGroupsSidebarItem
            singularTitle="My Squad"
            pluralTitle="My Squads"
            icon={GroupIcon}
          />
```

Reply:
Note i suggested eg `(props: { x`, while you have `({ props }: { props: { x`. The first is a single prop called x. Yours is a single prop called props which happens to be an object that has a field called x :)

Reply:
I'm super sorry, I feel like an idiot right now but I really dont' understand what I should be doing still. If I have a single prop then how would this still work?

          <MyGroupsSidebarItem
            singularTitle="My Squad"
            pluralTitle="My Squads"
            icon={GroupIcon}
          />


I'm so confused 😢

Reply:
OK, I've revert my changes to what I had to start, feel like this is a better place to try and improve from then with this props monster I ended up with.

I want to follow your suggestion I just think I'm missing something simple due to my lack of experience

Reply:
The confusion arises from mixing up the language - what a "prop" is, vs what an "argument" is. A function component that takes three props, still has just one argument like all other components do. It's a function that's called with one argument (often named `props` but that's not relevant actually), which is an object with three fields.

The jsx code 

```jsx
      <MyGroupsSidebarItem
        singularTitle="My Squad"
        pluralTitle="My Squads"
        icon={GroupIcon}
      />
```

is converted to the function call

```js
MyGroupsSidebarItem({
  singularTitle: "My Squad",
  pluralTitle: "My Squads",
  icon: GroupIcon,
})
```

(note the single object as a single argument to the function, with the three prop fields in it), not

```js
MyGroupsSidebarItem("My Squad", "My Squads", GroupIcon)
```

...which would have been three arguments.

And what your function does is to unpack that single first argument again, to extract the three fields out of it.

Reply:
But to be concrete: I think your code looks ok now. BUT the changeset here still has the old `props={{` syntax. So if you change it to

```
      <MyGroupsSidebarItem
        singularTitle="My Squad"
        pluralTitle="My Squads"
        icon={GroupIcon}
      />
```

instead, you're good to go